lib/view/login_screen.dart

```diff
@@ -0,0 +1,116 @@
+class LoginScreen extends StatelessWidget {
+  final AuthenticationController authController;
+
+  LoginScreen({required this.authController});
+
+  @override
+  Widget build(BuildContext context) {
+    TextEditingController emailController = TextEditingController();
+    TextEditingController passwordController = TextEditingController();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Log In'),
+      ),
+      body: Padding(
+        padding: EdgeInsets.all(16),
+        child: Column(
+          children: [
+            TextFormField(
+              controller: emailController,
+              decoration: InputDecoration(labelText: 'Email'),
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter your email';
+                }
+                if (!RegExp(r'^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$')
+                    .hasMatch(value)) {
+                  return 'Please enter a valid email';
+                }
+                return null;
+              },
+            ),
+            TextFormField(
+              controller: passwordController,
+              decoration: InputDecoration(labelText: 'Password'),
+              obscureText: true,
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter your password';
+                }
+                if (value.length < 8) {
+                  return 'Password must be at least 8 characters long';
+                }
+                if (!RegExp(
+                        r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$')
+                    .hasMatch(value)) {
+                  return 'Password must contain at least one uppercase letter, one lowercase letter, one digit, and one special character';
+                }
+                return null;
+              },
+            ),
+            SizedBox(height: 20),
+            ElevatedButton(
+              onPressed: () async {
+                if (emailController.text.isNotEmpty &&
+                    passwordController.text.isNotEmpty) {
+                  AuthenticationModel authModel = AuthenticationModel(
+                    email: emailController.text.trim(),
+                    password: passwordController.text.trim(),
+                  );
+                  try {
+                    await authController.login(
+                        authModel.email, authModel.password);
+                    Navigator.pushReplacement(
+                      context,
+                      MaterialPageRoute(
+                          builder: (context) => HomePage(
+                                authController: authController,
+                              )),
+                    );
+                  } catch (e) {
+                    ScaffoldMessenger.of(context).showSnackBar(
+                      SnackBar(
+                        content: Text(e.toString()),
```

Comment:
Maybe we need to print the error message in a better format

instead of "[firebase_auth/invalid-email]"

Reply:
sure but we have to consider every error code

Reply:
If you can ignore the first part of the error message, will be okay for now just remove "[firebase_auth/invalid-email]"